networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/peers/Eth2PeerManager.java

```diff
@@ -105,7 +111,49 @@ public static Eth2PeerManager create(
         peerValidatorFactory,
         attestationSubnetService,
         rpcEncoding,
-        eth2RpcPingInterval);
+        eth2RpcPingInterval,
+        eth2StatusUpdateInterval);
+  }
+
+  private void setUpPeriodicTasksForPeer(Eth2Peer peer) {
+    SafeFuture<Void> periodicStatusUpdateTask = periodicallyUpdatePeerStatus(peer);
+    SafeFuture<Void> periodicPingTask = periodicallyPingPeer(peer);
+    peer.subscribeDisconnect(() -> periodicStatusUpdateTask.cancel(false));
+    peer.subscribeDisconnect(() -> periodicPingTask.cancel(false));
+  }
+
+  @SuppressWarnings("FutureReturnValueIgnored")
```

Comment:
I misunderstood this when I read the ping PR, but we shouldn't be ignoring futures here.  If the `peer:sendStatus` call fails (ie we don't get a response from them or it's invalid) - what should we do?  Feels like we should disconnect if we get too many of those but definitely should at least report it at debug level.

Reply:
@Nashatyrev has a PR up that counts the number of outstanding requests and disconnects at a certain point. I think the future value still gets ignored in Anton's PR, but we count the number of outstanding requests instead. I think the same approach can be applied to periodically updating peer status.

Reply:
So, I'm probably going to merge it in and make another ticket for dealing with outstanding Status requests. (not to step Anton's toes even further)

Reply:
Also @Nashatyrev, this made me think, and am now curious, if instead of specifically counting outstanding PING requests (in this PR https://github.com/PegaSysEng/teku/pull/1779), if we should count all outstanding requests by type in the `Eth2Peer.sendRequest` method. It's probably important for us to know if the peer is not responding to requests of any type.

Reply:
I've also added `.finish()` handler to deal with success and failure by logging some information.

Reply:
`Ping` is sent with a predefined period so when missing some ping responses means that some significant time have passed without responses. While you may potentially do a lot of concurrent requests in say 10ms and all of them would be pending for responses

Reply:
I understand. So I think that check would need to be specific to each Eth2Peer.